create_hard_negatives_v2.py

```diff
@@ -820,15 +824,19 @@ def main(
                     if all_similar_acts_BL[i, 0] == -1:
                         continue
                     token_activations = []
+                    tokens: list[str] = []
                     max_act = 0
                     acts_L = all_similar_acts_BL[i, :].tolist()
-                    for j, token in enumerate(hard_negative_tokens):
+                    non_special_tokens = [token for token in hard_negative_tokens if token not in special_tokens]
+                    for j, token in enumerate(non_special_tokens):
                         if token in special_tokens:
                             continue
                         act = acts_L[j]
                         max_act = max(max_act, act)
-                        token_activations.append(TokenActivationV2.model_construct(s=token, act=act, pos=j))
-                    tokens = [act.s for act in token_activations]
+                        # only save if act > 0 for space reasons
+                        if act > 0.0:
+                            token_activations.append(TokenActivationV2.model_construct(s=token, act=act, pos=j))
+                        tokens.append(token)
```

Comment:
@adamkarvonen oops. I wasn't saving the tokens that are not activated. (We still continue to not save the activations of **non-activated tokens,** )